File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
bool cmp(string a, string b){
    return a + b > b + a;
}
string solution(vector<int> numbers) {
    string answer = "";
    vector<string> s;
    
    for(auto num : numbers){
        s.push_back(to_string(num));
    }
    sort(s.begin(), s.end(), cmp);
    if(s.front() == "0") return "0";
    
    
    for(auto ss: s){
        answer += ss;
    }
    return answer;
}

File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        List<String> list = new ArrayList<>();
        
        for(int i = 0; i < numbers.length;i++){
            list.add(String.valueOf(numbers[i]));
        }
        
        Collections.sort(list, new Comparator<String>(){
         @Override
         public int compare(String s1, String s2){
              return (s2+s1).compareTo(s1+s2);
        }
        });
        
        for(String s : list){
            answer += s;
        }
        
        if(list.get(0).equals("0")) answer = "0";
        return answer;
    }
}